pingouin/tests/test_multivariate.py

```diff
@@ -66,3 +68,30 @@ def test_multivariate_ttest(self):
         stats = multivariate_ttest(X_na, Y, paired=True)
         assert stats.at['hotelling', 'df1'] == 3
         assert stats.at['hotelling', 'df2'] == X.shape[0] - 1 - X.shape[1]
+
+    def test_box_m(self):
+        """Test function box_m.
+        Tested against the R package biotools (iris dataset).
+        """
+        
+        iris = datasets.load_iris()
+        df = pd.DataFrame(data= np.c_[iris['data'], iris['target']],
+                            columns= iris['feature_names'] + ['target'])
+
+        iris = datasets.load_iris()
+        df = pd.DataFrame(data= np.c_[iris['data'], iris['target']],
+                            columns= iris['feature_names'] + ['target'])
+        stats = box_m(df,vars=['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)'],group='target')
+        
+        # test for generic box_m
+        ##calculate covariance matrices
+        #cov_target0 = df[df['target']==0].iloc[:,:4].cov().values
+        #cov_target1 = df[df['target']==1].iloc[:,:4].cov().values
+        #cov_target2 = df[df['target']==2].iloc[:,:4].cov().values
+        #calculate the sample size for each 'group'
+        #sizes = [df[df['target']==0].shape[0],df[df['target']==1].shape[0],df[df['target']==2].shape[0]]
+        #stats = box_m(np.array([cov_target0,cov_target1,cov_target2]),sizes)
+        assert round(stats.at["Box's M", 'Chi2'], 5) == 140.94305
```

Comment:
Why is the Chi2 value rounded to 5? Does the output of R have more float precision?

Reply:
The precision of the output of box's M depends on the package of the function (i.e. MVTests, biotools, etc.) For example, in biotools, the Chi2 value is rounded to 2.